File: backend/src/main/java/br/com/julia/rgsurfschool/domain/model/Loja.java
package br.com.julia.rgsurfschool.domain.model;

import br.com.julia.rgsurfschool.domain.enums.Condicao;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity(name = "tb_loja")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
public class Loja {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    @Column(nullable = false)
    String nome;

    @Column(name = "qtd_estoque", nullable = false)
    Integer qtdEstoque;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    Condicao condicao;

    @Column(nullable = false, precision = 10, scale = 2)
    BigDecimal preco;

    @Column(nullable = false, precision = 10, scale = 2)
    BigDecimal custo;

    @Column(name = "data_aquisicao", nullable = false)
    LocalDate dataAquisicao;

    @Column(nullable = false)
    String fornecedor;

}

File: frontend/types/loja.ts
export enum Condicao {
  EXCELENTE = "EXCELENTE",
  BOM = "BOM",
  SEMINOVO = "SEMINOVO",
}

export interface LojaRequest {
  nome: string;
  qtdEstoque: number;
  condicao: Condicao;
  preco: number;
  custo: number;
  dataAquisicao: string; // ISO date string
  fornecedor: string;
}

export interface LojaResponse {
  id: number;
  nome: string;
  qtdEstoque: number;
  condicao: Condicao;
  preco: number;
  custo: number;
  dataAquisicao: string; // ISO date string
  fornecedor: string;
}
